openvdb_houdini/SOP_OpenVDB_Merge.cc

```diff
@@ -0,0 +1,296 @@
+// Copyright Contributors to the OpenVDB Project
+// SPDX-License-Identifier: MPL-2.0
+
+/// @file SOP_OpenVDB_Merge.cc
+///
+/// @author Dan Bailey
+///
+/// @brief Merge OpenVDB grids.
+
+#include "SOP_OpenVDB_Merge.proto.h"
+
+#include <PRM/PRM_TemplateBuilder.h>
+#include <UT/UT_DSOVersion.h>
+
+
+class SOP_OpenVDB_Merge : public SOP_Node
+{
+public:
+    static PRM_Template *buildTemplates();
+    static OP_Node *myConstructor(OP_Network *net, const char *name, OP_Operator *op)
+    {
+        return new SOP_OpenVDB_Merge(net, name, op);
+    }
+
+    static const UT_StringHolder theSOPTypeName;
+
+    virtual const SOP_NodeVerb *cookVerb() const final;
+
+protected:
+    SOP_OpenVDB_Merge(OP_Network *net, const char *name, OP_Operator *op)
+        : SOP_Node(net, name, op) { }
+
+    virtual ~SOP_OpenVDB_Merge() {}
+
+    virtual OP_ERROR cookMySop(OP_Context &context) final
+    {
+        return cookMyselfAsVerb(context);
+    }
+};
+
+
+const UT_StringHolder SOP_OpenVDB_Merge::theSOPTypeName("DW_OpenVDBMerge");
+
+
+void
+newSopOperator(OP_OperatorTable *table)
+{
+    table->addOperator(new OP_Operator(
+        SOP_OpenVDB_Merge::theSOPTypeName,
+        "VDB Merge",
+        SOP_OpenVDB_Merge::myConstructor,
+        SOP_OpenVDB_Merge::buildTemplates(),
+        1,
+        OP_MULTI_INPUT_MAX,
+        nullptr,
+        0));
+}
+
+
+static const char *theDsFile = R"THEDSFILE(
+{
+    name        parameters
+
+    inputlabel  1   "VDBs To Merge"
+
+    parm {
+        name    "group"
+        label   "Group"
+        type    string
+        default { "" }
+    }
+    parm {
+        name    "vdbpointsgroup"
+        label   "VDB Points Group"
+        type    string
+        default { "" }
+    }
+    parm {
+        name    "collation"
+        label   "Collation"
+        type    ordinal
+        default { "nameclassandtype" }
+        menu {
+            "nameclassandtype"  "Grid Name, Grid Class and Value Type"
+            "nameandclass"      "Grid Name and Grid Class"
+            "classandtype"      "Grid Class and Value Type"
+        }
+    }
+    parm {
+        name    "usereferencevdb"
+        label   "Use First Grid as Reference VDB Only"
```

Comment:
First VDB Is Only for Reference
(The left hand is only for support!)
Grid -> VDB
Imply the Use with the toggle.